src/CoreBundle/Response/ResponseVariableReplacerInterface.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Chameleon System (https://www.chameleonsystem.com).
+ *
+ * (c) ESONO AG (https://www.esono.de)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ChameleonSystem\CoreBundle\Response;
+
+use ChameleonSystem\CoreBundle\Security\AuthenticityToken\TokenInjectionFailedException;
+
+interface ResponseVariableReplacerInterface
+{
+    /**
+     * Add a variable that can then be replaced by replaceVariables(). Note that variables MUST alreday be escaped by
+     * the caller, as implementations of this interface do not know the context in which the variables are used and are
+     * therefore unable to perform the correct escaping.
```

Comment:
In the light of this comment we could discuss the name of the interface as `Response` kind of sets a context already.

Reply:
The name was chosen on purpose - the service adds the authenticity token (and throws a TokenInjectionFailedException on failure), which limits its scope to responses. 

Also having an addVariable() method in the interface does only make sense if the service is called on multiple independent pieces of client code. I wouldn't add the method on any generic replacer service (but leave this responsibility to implementations), so this also emphasizes the strategic (as opposed to generic) use of the interface.

Of course we can discuss if the name can be improved, but I wouldn't remove the purpose indicator.